src/pages/index.js

```diff
@@ -23,45 +34,172 @@ import {
   editPopupSelector,
   nameSelector,
   jobSelector,
-  initialCards,
   cardWrapperSelector,
 } from "../utils/constants.js";
 import UserInfo from "../components/UserInfo.js";
+import PopupWithConfirmation from "../components/PopupWithConfirmation.js";
 
+// elements
+let cardList;
+const deleteCardPopup = new PopupWithConfirmation(deleteCardSelector);
+const imageInfo = new PopupWithImage(picPopupSelector);
+const userProfile = new UserInfo({ nameSelector, jobSelector, avatarSelector });
+const addCardPopup = new PopupWithForm(
+  addCardPopupSelector,
+  handleAddCardSubmit
+);
+const editPopup = new PopupWithForm(editPopupSelector, handleProfileSubmit);
+const avatarEditPopup = new PopupWithForm(
+  avatarEditSelector,
+  handleAvatarSubmit
+);
+
+// api
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "62fbc1de-42d6-41b7-8bb5-5b0736932e96",
+    "Content-Type": "application/json",
+  },
+});
+
+// promise.all
+api
+  .getCardAndUserInfo()
+  .then(([serverCardList, serverUserInfo]) => {
+    // load cards from server
+    cardList = new Section(
+      {
+        items: serverCardList,
+        renderer: (item) => {
+          const cardItem = creatCard(item);
+          cardList.addInitialItem(cardItem);
+        },
+      },
+      cardWrapperSelector
+    );
+    // add initial cards to DOM
+    cardList.renderItems(serverCardList);
+    //likeBtn active if isLiked=true
+
+    // load user bio from server
+    const name = serverUserInfo.name;
+    const job = serverUserInfo.about;
+    const avatar = serverUserInfo.avatar;
+    // add user bio to DOM
+    userProfile.setUserInfo(name, job);
+    userProfile.setAvatar(avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
 // handler
+
 function handleAssignEditInput() {
   const userProfileData = userProfile.getUserInfo();
   editPopup.setInputValues(userProfileData);
 }
 
+function handleAvatarSubmit(inputObj) {
+  avatarEditPopup.showUploadingState();
+  const avatarLink = inputObj[avatarLinkInputId];
+  api
+    .updateProfileImage(avatarLink)
+    .then(avatarEditPopup.showUploaded())
+    .catch((err) => {
+      console.error(err);
+    });
+  avatarImg.src = avatarLink;
+  avatarEditPopup.close();
+  formValidators[avatarFormId].disableSubmitButton();
+  avatarEditPopup.reset();
+}
+
 function handleProfileSubmit(inputObj) {
-  userProfile.setUserInfo(
-    inputObj[editInputTitleId],
-    inputObj[editInputDescriptionId]
-  );
+  const nameEl = inputObj[editInputTitleId];
+  const jobEl = inputObj[editInputDescriptionId];
+  editPopup.showUploadingState();
+  api
+    .updateUserInfo({
+      editFormNameInput: nameEl,
+      editFormAboutInput: jobEl,
+    })
+    .then(editPopup.showUploaded())
+    .catch((err) => {
+      console.error(err);
+    });
+  // sent inputs to server
+  userProfile.setUserInfo(nameEl, jobEl);
```

Comment:
Needs correction: please update the info, reset and close the form only after the successful api response in `then` block otherwise the info will be updated and the form will close even if an error occurred